spaceapi/src/lib.rs

```diff
@@ -8,11 +8,35 @@ use rustc_serialize::json::{Json, ToJson};
 /// An Optional value can contain Optional::Some<T> or Optional::Absent.
 /// It is similar to an Option, but Optional::Absent means it will be
 /// omitted when serialized.
+#[derive(Debug, Copy, Clone)]
 pub enum Optional<T> {
     Value(T),
     Absent,
 }
 
+impl<T> Optional<T> {
+    pub fn map_or<U, F: FnOnce(T) -> U>(self, def: U, f: F) -> U {
+        match self {
+            Optional::Value(v) => f(v),
+            Optional::Absent => def,
+        }
+    }
```

Comment:
Also the difference in JSON between ``null`` and absent is not clear at all:
http://stackoverflow.com/questions/15686995/should-null-values-be-included-in-json-responses-from-a-rest-api
https://groups.google.com/forum/?fromgroups=#!topic/api-craft/MrEixUwOCWA